examples/cmake_hello_world_variant/BUILD

```diff
@@ -1,10 +1,27 @@
 load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
 
+# TODO: This should not be necessary but there appears to be some inconsistent
+# behavior with the use of `constraint_value`s in `select` statements. A support
+# thread was started at the end of https://github.com/bazelbuild/bazel/pull/12071
+# Once it is possible to replace `:windows` with `@platforms//os:windows` that
+# should be done for this file
+config_setting(
```

Comment:
If the changes here otherwise look good, I'll create a new issue for this. Hopefully someone knows what's going on here.